src/shared/ask-password-api.c

```diff
@@ -168,7 +168,16 @@ static int ask_password_keyring(const AskPasswordRequest *req, AskPasswordFlags
         if (r < 0)
                 return r;
 
-        return retrieve_key(serial, ret);
+        _cleanup_strv_free_erase_ char **l = NULL;
+        r = retrieve_key(serial, &l);
+        if (r < 0)
+                return r;
+
+        if (strv_isempty(l))
+                return log_debug_errno(SYNTHETIC_ERRNO(ENOKEY), "Found an empty password from keyring.");
```

Comment:
i thnk a similar check is still missing in ask_password_plymouth(). Hopefully plymouth won't send us an empty password list, but better check that too.

Reply:
I noticed that, but it is currently used by only tty-ask-password.c, and the caller checks if the strv is empty. So, I did not touch that part.

Now, for consistency, `_plymouth()` also checks the strv, and dropped the check in tty-ask-password.